contracts/contracts/ERC677InitializableTokenV2.sol

```diff
@@ -0,0 +1,47 @@
+pragma solidity 0.4.24;
+
+import "./ERC677InitializableToken.sol";
+
+contract ERC677InitializableTokenV2 is ERC677InitializableToken {
+    mapping(address => bool) allowTransferCallback;
+
+    function transfer(address to, uint256 value) public returns (bool)
+    {
+        require(superTransfer(to, value), "failed superTransfer");
+        fundReceiver(to);
+
+        if (isContract(to)) {
+            if (to == bridgeContract || allowTransferCallback[to]) {
+                if (!triggerTransferCallback(to, value, new bytes(0))) {
+                    revert("Invoke transfer callback failed");
+                }
+            }
+        }
+        return true;
+    }
+
+    function enableTransferCallback(address addr) public onlyOwner {
+        allowTransferCallback[addr] = true;
+    }
+
+    function disableTransferCallback(address addr) public onlyOwner {
+        delete allowTransferCallback[addr];
+    }
+
+    function triggerTransferCallback(address addr, uint value, bytes data) internal returns (bool) {
+        return addr.call(
+            abi.encodeWithSignature(
+                "onTokenTransfer(address,uint256,bytes)",
+                msg.sender,
+                value,
+                data
+            )
+        );
+    }
+
+    function isContract(address addr) private view returns (bool) {
```

Comment:
Since `triggerTransferCallback` is set to internal, might as well set this to internal too?

Reply:
no problem!

Reply:
It seems not able to change visibility directly:

```
.../thunder_bridge/contracts/contracts/ERC677InitializableTokenV2.sol:42:5: TypeError: Overriding function visibility differs.
    function isContract(address addr) internal view returns (bool) {
    ^ (Relevant source part starts here and spans across multiple lines).
.../thunder_bridge/contracts/contracts/ERC677InitializableToken.sol:132:5: Overriden function is here:
    function isContract(address _addr)
    ^ (Relevant source part starts here and spans across multiple lines).

```

so I renamed the method to `isCallable`

Reply:
Got it, learned a lesson!